soh/src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -631,7 +634,7 @@ void EnBox_UpdateSizeAndTexture(EnBox* this, GlobalContext* globalCtx) {
         }
     }
 
-    if (globalCtx->sceneNum != SCENE_TAKARAYA && (cvar == 1 || cvar == 3)) {
+    if (globalCtx->sceneNum != SCENE_TAKARAYA && (cvar == 1 || cvar == 3) && ((agonyCVar > 0 && stoneCheck) | agonyCVar == 0)) {
```

Comment:
Looks like you missed one spot this condition needs to be added, down below at 665